packages/react-dom/src/events/EnterLeaveEventPlugin.js

```diff
@@ -169,11 +169,18 @@ const EnterLeaveEventPlugin = {
 
     accumulateEnterLeaveDispatches(leave, enter, from, to);
 
-    if (nativeEvent === lastNativeEvent) {
-      lastNativeEvent = null;
+    if (seenNativeEvents.has(nativeEvent)) {
+      if (seenNativeEvents instanceof Set) {
```

Comment:
Can we get away with not performing the `instanceof` check each time? We should know at the point `seenNativeEvents` is defined whether or not we're working with a `Set` or `WeakSet`